go/test/endtoend/reparent/utils/utils.go

```diff
@@ -863,3 +863,22 @@ func WaitForTabletToBeServing(ctx context.Context, t *testing.T, clusterInstance
 		t.Fatal(err.Error())
 	}
 }
+
+// WaitForQueryWithStateInProcesslist waits for a query to be present in the processlist with a specific state.
+func WaitForQueryWithStateInProcesslist(ctx context.Context, t *testing.T, tablet *cluster.Vttablet, sql, state string, timeout time.Duration) {
+	require.Eventually(t, func() bool {
+		qr := RunSQL(ctx, t, "select Command, State, Info from information_schema.processlist", tablet)
```

Comment:
Just a note that we might want to use `performance_schema.processlist` instead as it will show the last statement executed in the session/connection if there is not a current one. I'm not sure that we'd want that here, so just noting it.